### DISCUSSION TITLE
Web-UI Container not starting on latest main

### DISCUSSION THREAD

**Discussion by Gerthum** [open-webui/open-webui#743] [General]
# Bug Report

Getting command not found when trying to run the docker container on windows. It was working yesterday I think it was before the latest commit. 

## Description

`start.sh: line 2: $'\r': command not found
start.sh: line 4: cd: $'/app/backend\r': No such file or directory
start.sh: line 4: $'exit\r': command not found
start.sh: line 5: $'\r': command not found
start.sh: line 7: $'\r': command not found
start.sh: line 23: syntax error: unexpected end of file
start.sh: line 2: $'\r': command not found
start.sh: line 4: cd: $'/app/backend\r': No such file or directory
start.sh: line 4: $'exit\r': command not found
start.sh: line 5: $'\r': command not found
start.sh: line 7: $'\r': command not found
start.sh: line 23: syntax error: unexpected end of file`

**Bug Summary:**
[Provide a brief but clear summary of the bug]

Docker container not starting.

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

Windows 10 running Rancher Desktop

## Reproduction Details

Run: docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v ollama-webui:/app/backend/data --name ollama-webui --restart always ghcr.io/ollama-webui/ollama-webui:main

Container will keep crashing

**Confirmation:**

- [x ] I have read and followed all the instructions provided in the README.md.
- [x ] I have reviewed the troubleshooting.md document.
- [x ] I have included the browser console logs.
- [x ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by tjbck** [comment]
Hi, I'm unable to reproduce the issue on my machines, could provide more detailed steps to reproduce?

https://github.com/ollama-webui/ollama-webui/pull/723 might be the culprit here but I'm unable to verify.

**Comment by Gerthum** [comment]
Thanks @tjbck - Are you testing on Windows with docker? 

All I do is run: docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v ollama-webui:/app/backend/data --name ollama-webui --restart always ghcr.io/ollama-webui/ollama-webui:main

Then the container doesn't start. 

Is there a way to pull an image before the last merge? Then I can test and see if it resolve the issue.

**Comment by justinh-rahb** [comment]
> Is there a way to pull an image before the last merge? Then I can test and see if it resolve the issue.

@Gerthum this should be the commit just before #723:

```bash
docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v ollama-webui:/app/backend/data --name ollama-webui --restart always ghcr.io/ollama-webui/ollama-webui:main@sha256:4bbc4bb15657c9bdc9d7736898356d14a0a91dadb3dade0774d6b0558eb88fc2
```

@tjbck perhaps it's time to consider modifying the Docker workflow to created versioned tags. Even if we just used an auto-incrementing build number or the commit hash that triggered the action for now, would make pulling specific docker images easier than hunting down the sha256 hash. **ETA: I've got something that will work for this #739.**

<img width="754" alt="Screenshot 2024-02-15 at 12 02 02 AM" src="https://github.com/ollama-webui/ollama-webui/assets/52832301/dcace3d8-f20a-4222-9631-1b2b6b5aed50">
<img width="754" alt="Screenshot 2024-02-15 at 12 02 16 AM" src="https://github.com/ollama-webui/ollama-webui/assets/52832301/54f75b31-a06b-43f8-af4c-b303fd4ca6ac">

**Comment by dangminhduc1101** [comment]
You might want to `docker rm -f ollama-webui` and `docker pull ghcr.io/ollama-webui/ollama-webui:main` first, then `docker run -d -p 3000:8080 --add-host=host.docker.internal:host-gateway -v ollama-webui:/app/backend/data --name ollama-webui --restart always ghcr.io/ollama-webui/ollama-webui:main`

**Comment by jannikstdl** [comment]
Yes, this issue should be fixed in #723

Repull the latest Image and try again.

**Comment by tjbck** [comment]
Just tested with two of my windows machines, still unable to reproduce the issue on my side. Keep us updated!
